### ISSUE TITLE
Baseline creation throws an error when I skip a test in my test command

### ISSUE THREAD

**Issue by cedaniel200** [sixty-north/cosmic-ray#549]
I have the following configuration file:

```
[cosmic-ray]
module-path = ["application", "domain", "entrypoints", "infraestructure", "commons.py"]
timeout = 10.0
excluded-modules = []
test-command = "python3 -m pytest -k 'not integration' ../../tests/src/calculator"

[cosmic-ray.distributor]
name = "local"
```
This command allows me to skip some integration tests, said test runs correctly on my machine:
Command: `python3 -m pytest -k 'not integration' ../../tests/src/calculator`
output:
![image](https://github.com/user-attachments/assets/23fe8e73-5309-4d3e-8709-3641eb350d44)

When I try to run the baseline command c`osmic-ray --verbosity=INFO baseline mt-calculator.toml` gives as output:
![image](https://github.com/user-attachments/assets/38ea2c87-659c-41cb-a136-e30074186ab8)

If I remove the -k 'not integration'  baseline command everything works correctly, but the contract/integration tests would be running and this is behavior that I do not want.

Any ideas on how to solve it, thanks in advance

**Comment by abingham** [comment]
I'm able to replicate this. It looks like the problem is in `cosmic_ray.testing.run_tests()`, line 45, where we call `split()` on the test command to pass it into `subprocess.run()`. This is to naive, resulting in "'not" and "integration'" being passed as individual arguments to the subprocess; we should instead be passing "'not integration'" as a single argument. I'll look into this.

**Comment by abingham** [comment]
Looks like we need to use [`shlex.split`](https://docs.python.org/3/library/shlex.html#shlex.split).

**Comment by abingham** [comment]
This should be fixed in 8.3.15, but let me know if it's still a problem. Thanks for reporting this!
